config/fastly/snippets/client_geolocation.vcl

```diff
@@ -1,5 +1,18 @@
+sub vcl_init {
+  table active_geofencing_regions {
+    // This (empty) table is managed independently of the deployed configuration as an edge dictionary.
+
+    // This accomplishes a couple of things:
+    // 1. It greatly reduces the number of variants that need to be cached
+    // 2. It lets us enable more granular caching (for big campaigns and/or customers)
+    //    without having to redeploy the app.
+    // 3. It lets us turn off caching entirely fairly easily (by emptying the table)
+  }
+}
```

Comment:
I feel like some more context on this would be useful (even with the docs):

I have been struggling with coming up with a way to cache the `/billboards/:area` endpoint that doesn't fall into one of two extremes, either:
- not being able to cache anything at all (so signed out users effectively cannot be targeted by location), or
- caching too many location variants, which is potentially expensive (both in Fastly charges and in server time due to cache misses) and may not even be possible as we expand to more countries (there are only so many variants that Fastly will store).

The solution I have here is to make the regions with cached billboards content dynamic; the table definition here creates a [Fastly edge dictionary](https://developer.fastly.com/learning/concepts/edge-state/dynamic-config/#edge-dictionaries) which can be updated independently of the app deploy and rolls out in less than a minute. We can add and remove specific locations that we could like to cache responses for to this dictionary instead of having trying to have baked-in support for M countries x N regions

As an example we could add `"CA": "CA"` and `"US": "US"` as edge dictionary entries to start (which would enable caching at a country level for those countries - returning & caching billboards that target the whole of those countries, but not specific states/provinces). Then we could add more countries we would want to cache billboard content for as we expand the feature (which may not be the same as all the countries we support targeting in). Additionally, to e.g. support a specific event (like say an in-person hackathon in New York) we could briefly add `"CA-NY": "CA-NY"` to the dictionary to give the targeted billboard greater reach until its campaign is over.

cc @benhalpern